lib/src/lifecycle_module.dart

```diff
@@ -130,12 +156,28 @@ abstract class LifecycleModule extends Object with Disposable {
       _didUnloadChildModuleSubscriptions = {};
 
   /// Whether the module is currently loaded.
-  bool get isLoaded => _isLoaded;
+  bool get isLoaded => _state == LifecycleState.loaded;
+
+  /// Whether the module is currently loading.
+  bool get isLoading => _state == LifecycleState.loading;
+
+  /// Whether the module is currently resuming.
+  bool get isResuming => _state == LifecycleState.resuming;
 
   /// Whether the module is currently suspended.
-  ///
-  /// This will always be false when the module is not loaded.
-  bool get isSuspended => _isSuspended;
+  bool get isSuspended => _state == LifecycleState.suspended;
+
+  /// Whether the module is currently suspending.
+  bool get isSuspending => _state == LifecycleState.suspending;
+
+  /// Whether the module is currently unloaded.
+  bool get isUnloaded => _state == LifecycleState.unloaded;
+
+  /// Whether the module is currently unloading.
+  bool get isUnloading => _state == LifecycleState.unloading;
+
+  /// The current lifecycle state of the module.
+  LifecycleState get state => _state;
```

Comment:
The current need to expose this field is to simplify tests. It is easy to write a utility function to relive the need of this property.

I would argue that there is no harm in exposing both the `state` property as well as the `is<LifecycleState>` utility functions as long as the latter is always written in terms of the former.